src/app/(pages)/educations/_components/List/EducationsListTable.tsx

```diff
@@ -6,36 +6,68 @@ import {
   TableHeader,
   TableRow,
 } from "~/components/organisms/Table";
-import { Pencil } from "lucide-react";
 import PaginationComponent from "~/components/organisms/pagination/PaginationComponent";
-import { type FC } from "react";
+import { useState, useEffect, type FC } from "react";
 import type { FindEducationReturnDTO } from "~/server/services/education/education.repository";
 import { translateEducationType } from "~/app/(pages)/educations/utils";
 import { type EducationType } from "~/server/db/schema";
+import Link from "next/link";
+import { api } from "~/trpc/react";
+import { Trash2, Pencil } from "lucide-react";
 
 type Props = {
   data?: FindEducationReturnDTO[];
   totalPageNumber: number;
 };
 
-const EducationsListTable: FC<Props> = ({ data, totalPageNumber }) => {
+const EducationsListTable: FC<Props> = ({
+  data: initialData,
+  totalPageNumber,
+}) => {
+  const [data, setData] = useState<FindEducationReturnDTO[]>([]);
+
+  useEffect(() => {
+    if (initialData) {
+      setData(initialData);
+    }
+  }, [initialData]);
+
+  const deleteEducation = api.education.deleteById.useMutation();
+
+  const handleDelete = async (id: string) => {
+    const confirmed = window.confirm(
```

Comment:
What do you think of using `<Modal />` here instead of this ugly default browser window and write a custom action?